xc/device.ml

```diff
@@ -34,27 +34,23 @@ open D
 
 (** Definition of available qemu profiles, used by the qemu backend implementations *)
 module Profile = struct
-  type t = Qemu_trad | Qemu_upstream_compat | Qemu_upstream | Qemu_upstream_uefi [@@deriving rpcty]
-  let fallback = Qemu_trad
-  let all = [ Qemu_trad; Qemu_upstream_compat; Qemu_upstream ]
+  type t = Qemu_none | Qemu_upstream_compat | Qemu_upstream | Qemu_upstream_uefi [@@deriving rpcty]
+  let fallback = Qemu_upstream_compat
+  let all = [ Qemu_none; Qemu_upstream_compat; Qemu_upstream ]
   module Name = struct
+    let qemu_none            = "qemu-none"
     let qemu_trad            = "qemu-trad"
     let qemu_upstream_compat = "qemu-upstream-compat"
     let qemu_upstream        = "qemu-upstream"
     let qemu_upstream_uefi   = "qemu-upstream-uefi"
   end
   let wrapper_of = function
-    | Qemu_trad            -> !Resources.qemu_dm_wrapper
+    | Qemu_none            -> "/bin/false"
     | Qemu_upstream_compat -> !Resources.upstream_compat_qemu_dm_wrapper
     | Qemu_upstream        -> !Resources.upstream_compat_qemu_dm_wrapper
     | Qemu_upstream_uefi   -> !Resources.upstream_compat_qemu_dm_wrapper
-  let string_of  = function
-    | Qemu_trad              -> Name.qemu_trad
-    | Qemu_upstream_compat   -> Name.qemu_upstream_compat
-    | Qemu_upstream          -> Name.qemu_upstream
-    | Qemu_upstream_uefi     -> Name.qemu_upstream_uefi
   let of_string  = function
-    | x when x = Name.qemu_trad            -> Qemu_trad
+    | x when x = Name.qemu_trad            -> Qemu_upstream_compat
```

Comment:
Is this correct? Should this fail or result in a warning?

Reply:
A warning is a good idea. Whether it should fail or not we should discuss with @BobBall.
I'd be in faviour of just a warning because if we fail then it makes RPU more complicated: you can suddenly end up with VMs that you cannot migrate anymore if they got qemu-trad set explicitly as a profile.

Reply:
I think that we should warn here, and not fail. If we want to make things fail, we would ideally do that earlier, in xapi.